QuantConnect.IBAutomater/IBAutomater.cs

```diff
@@ -101,72 +107,104 @@ public IBAutomater(string ibDirectory, string ibVersion, string userName, string
 
-            var fileName = IsWindows ? "IBAutomater.bat" : "IBAutomater.sh";
-            var arguments = $"{_ibDirectory} {_ibVersion} {_userName} {_password} {_tradingMode} {_portNumber}";
+                var process = new Process
+                {
+                    StartInfo = new ProcessStartInfo(fileName, arguments)
+                    {
+                        RedirectStandardOutput = true,
+                        RedirectStandardError = true,
+                        UseShellExecute = false,
+                        WindowStyle = ProcessWindowStyle.Hidden,
+                        CreateNoWindow = true
+                    },
+                    EnableRaisingEvents = true
+                };
 
-            var process = new Process
-            {
-                StartInfo = new ProcessStartInfo(fileName, arguments)
+                process.OutputDataReceived += (sender, e) =>
                 {
-                    RedirectStandardOutput = true,
-                    RedirectStandardError = true,
-                    UseShellExecute = false,
-                    WindowStyle = ProcessWindowStyle.Hidden,
-                    CreateNoWindow = true
-                },
-                EnableRaisingEvents = true
-            };
+                    if (e.Data != null)
+                    {
+                        OutputDataReceived?.Invoke(this, new OutputDataReceivedEventArgs(e.Data));
+                    }
+                };
 
-            process.OutputDataReceived += (sender, e) =>
-            {
-                if (e.Data != null)
+                process.ErrorDataReceived += (sender, e) =>
                 {
-                    OutputDataReceived?.Invoke(this, new OutputDataReceivedEventArgs(e.Data));
-                }
-            };
+                    if (e.Data != null)
+                    {
+                        ErrorDataReceived?.Invoke(this, new ErrorDataReceivedEventArgs(e.Data));
+                    }
+                };
 
-            process.ErrorDataReceived += (sender, e) =>
-            {
-                if (e.Data != null)
+                process.Exited += (sender, e) =>
+                {
+                    Exited?.Invoke(this, new ExitedEventArgs(process.ExitCode));
+                };
+
+                var started = process.Start();
+                if (!started)
                 {
-                    ErrorDataReceived?.Invoke(this, new ErrorDataReceivedEventArgs(e.Data));
+                    throw new Exception("IBAutomater was unable to start the IBGateway process.");
                 }
-            };
 
-            process.Exited += (sender, e) =>
-            {
-                Exited?.Invoke(this, new ExitedEventArgs(process.ExitCode));
-            };
+                OutputDataReceived?.Invoke(this, new OutputDataReceivedEventArgs($"IBAutomater process started - Id:{process.Id}"));
 
-            process.Start();
+                _processId = process.Id;
```

Comment:
Maybe should we set `_processId` to 0 when we set starting to `true` so that we avoid any race condition